cpu/stm32/periph/rtc_all.c

```diff
@@ -372,12 +398,23 @@ void rtc_poweroff(void)
 
 void ISR_NAME(void)
 {
+#if !defined(CPU_FAM_STM32L5)
     if (RTC_REG_ISR & RTC_ISR_ALRAF) {
         if (isr_ctx.cb != NULL) {
             isr_ctx.cb(isr_ctx.arg);
         }
         RTC_REG_ISR &= ~RTC_ISR_ALRAF;
     }
     EXTI_REG_PR = EXTI_PR_BIT; /* only clear the associated bit */
+#else
+    if (RTC_REG_SR & RTC_SR_ALRAF) {
+        if (isr_ctx.cb != NULL) {
+        isr_ctx.cb(isr_ctx.arg);
+        }
+        stmclk_dbp_unlock();
+        RTC_REG_SCR = RTC_SCR_CALRAF;
+        stmclk_dbp_lock();
+    }
+#endif
```

Comment:
You mention this is not specified in the datasheet or errata, how did you come up to this fix? Do you mind adding a comment mentioning the reasoning?

Reply:
Found it surfing through `STM32` forum where a similar problem was adviced. https://community.st.com/s/question/0D50X00009XkgO1/stm32l4-hal-rtc-clear-alarm-flags

Reply:
Ahh that makes sense, when looking at the doc:

![image](https://user-images.githubusercontent.com/23060007/126165492-61b2421e-654d-4df1-b6b3-7be8729eb33c.png)

can you add some comment regarding this?

Reply:
Comments added!